lib/services/debug-service.ts

```diff
@@ -0,0 +1,65 @@
+import { platform } from "os";
+import { EventEmitter } from "events";
+import { CONNECTION_ERROR_EVENT_NAME } from "../constants";
+
+class DebugService extends EventEmitter {
+	constructor(private $devicesService: Mobile.IDevicesService,
+		private $androidDebugService: IPlatformDebugService,
+		private $iOSDebugService: IPlatformDebugService,
+		private $errors: IErrors,
+		private $hostInfo: IHostInfo,
+		private $devicePlatformsConstants: Mobile.IDevicePlatformsConstants) {
+		super();
+	}
+
+	public async debug(debugData: IDebugData, options: IDebugOptions): Promise<string> {
+		const device = this.$devicesService.getDeviceByIdentifier(debugData.deviceIdentifier);
+		const debugService = this.getDebugService(device);
+
+		debugService.on(CONNECTION_ERROR_EVENT_NAME, (e: Error) => this.emit(CONNECTION_ERROR_EVENT_NAME, e));
+
+		if (!device) {
+			this.$errors.failWithoutHelp(`Can't find device with identifier ${debugData.deviceIdentifier}`);
+		}
+
+		const debugOptions: IDebugOptions = _.merge({}, options || {});
```

Comment:
I believe you shouldn't worry if options is undefined, lodash works fine in these cases:
```TypeScript
const debugOptions: IDebugOptions = _.merge({}, options);
```